.github/workflows/ci.yaml

```diff
@@ -80,6 +80,40 @@ jobs:
         run: sudo make build
       - name: test-e2e
         run: sudo -E make test-e2e-for-awslogs # containerd interaction requires sudo and aws cloudwatch interaction requires passing env vars
+  e2e-tests-for-fluentd:
+    strategy:
+      fail-fast: false
+      matrix:
+        go: [ '1.20', '1.21' ]
+        os: [ ubuntu-latest ] # TODO: Add Windows e2e tests: https://github.com/aws/shim-loggers-for-containerd/issues/68
+    name: E2E tests / fluentd / ${{ matrix.os }} / Go ${{ matrix.go }}
+    runs-on: ${{ matrix.os }}
+    permissions:
+      id-token: write
+      contents: write
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v4
+        with:
+          go-version: ${{ matrix.go }}
+          cache: false
+      - name: install and start containerd
+        shell: bash
+        run: sudo scripts/install-containerd
+      - name: start fluentd local endpoint
+        shell: bash
+        run: |
+          : # not using github action env because env map cannot be defined in terms of other vars in the map. public.ecr.aws/docker/library/fluentd:v1.16-debian-1
+          FLUENTD_LOG_DIR=${GITHUB_WORKSPACE}/fluentd-logs
+          FLUENTD_PORT=24224
+          FLUENTD_IMAGE=public.ecr.aws/docker/library/fluentd:v1.16-debian-1
+          : # Fluentd container is not using root user so need 777 to make it writable. https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#docker-container-filesystem
+          sudo mkdir -m 777 $FLUENTD_LOG_DIR
```

Comment:
I'm not super familiar with how github actions work, will this log dir eventually get cleaned up or should we add some logic to clean it up after each test?

Reply:
Each workflow runs in a fresh environment so cleaning up doesn't matter for the basic scenario. 

It is nice to have and may be need for future complex scenarios. It is a bit complicated to clean up Fluentd because it will be not working if I delete the whole log file. (Maybe the Fluentd workflow assumes the existing file or existing history). We can add it when needed in the future.